Signal/src/view controllers/DebugUITableViewController.m

```diff
@@ -0,0 +1,271 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "DebugUITableViewController.h"
+#import "ThreadUtil.h"
+#import "Environment.h"
+#import <SignalServiceKit/TSThread.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class OWSTableItem;
+@class OWSTableSection;
+
+@interface OWSTableContents : NSObject
+
+@property (nonatomic) NSString *title;
+@property (nonatomic) NSMutableArray *sections;
+
+@end
+
+#pragma mark -
+
+@implementation OWSTableContents
+
+- (void)addSection:(OWSTableSection *)section {
+    OWSAssert(section);
+    
+    if (!_sections) {
+        _sections = [NSMutableArray new];
+    }
+    
+    [_sections addObject:section];
+}
+
+@end
+
+#pragma mark -
+
+@interface OWSTableSection : NSObject
+
+@property (nonatomic) NSString *title;
+@property (nonatomic) NSMutableArray *items;
+
+@end
+
+#pragma mark -
+
+@implementation OWSTableSection
+
++ (OWSTableSection *)sectionWithTitle:(NSString *)title
+                                items:(NSArray *)items {
+    OWSTableSection *section = [OWSTableSection new];
+    section.title = title;
+    section.items = [items mutableCopy];
+    return section;
+}
+
+- (void)addItem:(OWSTableItem *)item {
+    OWSAssert(item);
+    
+    if (!_items) {
```

Comment:
nit: better to initialize _items in `init` since we treat it as a nonnull property.